### PR TITLE
build: workflow to automate the PR creation

### LINKED ISSUE #357: Add sync workflow for config param names
Depends on https://github.com/ubiquity/ubiquibot/issues/327

We are in the process of an active development so we refactor the bot's [config](https://github.com/ubiquity/ubiquibot/blob/development/.github/ubiquibot-config.yml) param names occasionally.

The latest refactors (both implemented and planned):
- move from camel case to kebab case (implemented)
- refactor `chainId` to `networkId` (planned)

Updating config param names for all our [partners](https://github.com/ubiquity/devpool-directory/blob/development/projects.json) is tedious. We should automate this process.

What should be done:
- When https://github.com/ubiquity/ubiquibot/issues/327 is implemented create a github workflow with manual dispatch that takes the default yml config (from https://github.com/ubiquity/ubiquibot/issues/327) and creates PRs with updated config param names for [partner projects](https://github.com/ubiquity/devpool-directory/blob/development/projects.json)

So in the end the workflow should be able to create config update PRs which:
1. Add newly added bot's config params
2. Rename config params (example: `chain-id => evm-network-id`)

### PR DESCRIPTION
**Pull request by AnakinSkywalkeer**
Resolves #357

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Check out repository
+        uses: actions/checkout@v3
+
+      - name: Set up Git
+        run: |
+          git config --global user.name "${{ github.actor }}"
+          git config --global user.email "${{ github.actor }}@users.noreply.github.com"
+      
+      - name: Install jq
+        run: sudo apt-get -y install jq
```
Comment (rndquu) [inline .github/workflows/pr.yml:18]: Why do we need this step?

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      
+      - name: Update config params for partner projects
+        run: |
+          declare -a repoUrls=(
```
Comment (rndquu) [inline .github/workflows/pr.yml:22]: Partner URLs should be dynamically fetched from https://github.com/ubiquity/devpool-directory/blob/development/projects.json

File: .github/workflows/pr.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+          )
+          for url in "${repoUrls[@]}"
+          do
+            repoName=$(basename $url)
+            ownerName=$(echo $url | awk -F/ '{print $(NF-1)}')
+            git clone $url 
+            cd $repoName
+            defaultBranch=$(git remote show origin | awk '/HEAD branch/ {print $NF}')
+
+            cp ../.github/ubiquity-bot.yml .github/ubiquity-bot.yml
```
Comment (rndquu) [inline .github/workflows/pr.yml:44]: `cp` rewrites the config, we need it to be updated with new parameters

**Review by rndquu — CHANGES_REQUESTED** [review]
Pls fix:
1. [QA](https://github.com/ubiquity/ubiquibot#how-to-qa-any-additions-to-the-bot) is missing
2. The original issues describes that we should compare the [default config](https://github.com/ubiquity/ubiquibot/pull/383/files#diff-3dbd906a74e6e90c70b674f19ce3a371e7c77361c9de83dc8fdf9f02750caa8b) with a partner's config but [PR](https://github.com/ubiquity/ubiquibot/pull/383) for this feature is not yet merged

**Comment by pavlovcik** [comment]
Stale

**Pull request closed without merging**